app/models/concerns/decision_review_tasks_concern.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module DecisionReviewTasksConcern
+  extend ActiveSupport::Concern
+
+  def in_progress_tasks(
+    _sort_by: "",
+    sort_order: "desc",
+    _filters: []
+  )
+    Task.select(Arel.star)
+      .from(combined_decision_review_tasks_query)
+      .includes(*decision_review_task_includes)
+      .order(assigned_at: sort_order.to_sym)
+  end
+
+  def completed_tasks(
+    _sort_by: "",
+    sort_order: "desc",
+    _filters: []
+  )
+    tasks
+      .recently_completed
+      .includes(*decision_review_task_includes)
+      .order(closed_at: sort_order.to_sym)
+  end
+
+  private
+
+  def decision_review_task_includes
+    [:assigned_to, :appeal]
+  end
+
+  def issue_count
+    # Issue count alias for sorting and serialization
+    "COUNT(request_issues.id) AS issue_count"
+  end
+
+  def higher_level_reviews_on_request_issues
+    decision_reviews_on_request_issues(higher_level_review: :request_issues)
+  end
+
+  def supplemental_claims_on_request_issues
+    decision_reviews_on_request_issues(supplemental_claim: :request_issues)
+  end
+
+  def appeals_on_request_issues
+    decision_reviews_on_request_issues(ama_appeal: :request_issues)
+  end
+
```

Comment:
```suggestion

  # Specific case for BoardEffectuationGrantTasks
  def board_grant_effectuation_tasks
    Task.select(Task.arel_table[Arel.star], issue_count)
      .open
      .left_joins(ama_appeal: :request_issues)
      .where(Task.arel_table[:type].eq(BoardGrantEffectuationTask.name))
      .where(assigned_at: id)
      .group("tasks.id")
      .arel
  end
  
```
Here's the union version of this query. It's a bit faster but not by a significant margin whatsoever. It does clean up some of the where clause logic and will exactly mimic the original query even for no request issues even if that isn't a likely scenario. You can decide what to do after playing around with it